### DISCUSSION TITLE
Stream Query Results without RawQuery

### DISCUSSION THREAD

**Discussion by yjagota** [ravendb/ravendb#17020] [Q&A] [answered]
Hello,

The documentation of RavenDb gives the following example for [stream query results](https://ravendb.net/docs/article-page/5.4/csharp/client-api/session/querying/how-to-stream-query-results#stream-related-documents) with includes:

```
IRawDocumentQuery<AllDocsProjection> query = session
    .Advanced
    .RawQuery<AllDocsProjection>(@"from Orders as o 
                                   where o.ShipTo.City = 'London'
                                   load o.Company as c, o.Employee as e
                                   select {
                                       Order: o,
                                       Company: c,
                                       Employee: e
                                   }");

// Call 'Stream' to execute the query
IEnumerator<StreamResult<AllDocsProjection>> streamResults = session.Advanced.Stream(query);
```

But the same documentation says:

> The below example uses RawQuery.
> However, the same logic can be applied to a Query, DocumentQuery, or when querying an index.

But cannot find any way to do it statically without query strings. Is this doable or I have to use RawQuery?

**Comment by ayende** [comment]
Here is one such example:

https://github.com/ravendb/ravendb/blob/20464f7947e16803357a145d2e787f7ac0a13c7e/test/SlowTests/SlowTests/MailingList/Jalchr2.cs#L101

Can you explain what you are trying to do?

**Reply by yjagota** [reply]
I don't think this covers what I am asking.

Let's take the exact example in the documentation:
```
IRawDocumentQuery<AllDocsProjection> query = session
    .Advanced
    .RawQuery<AllDocsProjection>(@"from Orders as o 
                                   where o.ShipTo.City = 'London'
                                   load o.Company as c, o.Employee as e
                                   select {
                                       Order: o,
                                       Company: c,
                                       Employee: e
                                   }");
```

I can do this in Index Maps using LoadDocument. 

I can do this in regular queries with includes:
```
repo.Query<Order, OrderIndex>()
	.Where(x => x.ShipTo.City == "London")
	.Include<Company>(x => x.Company)
	.Include<Employee>(x => x.Employee)
	.ToList()
	
Company company = session.Load<Company>(order.Company);
Employee employee= session.Load<Employee>(order.Employee);
```

My query is how can I do it with streams, without using RawQuery:
```
query = repo.Query<Order, OrderIndex>()
	.Where(x => x.ShipTo.City == "London")
	[load o.Company as c, o.Employee as e] == ?? // How to I translate this into C# code without RawQuery?
	.ProjectInfo<AllDocsProjection>()

var results = repo.Stream(query);
while (await results.MoveNextAsync())
{
	...
}
```
Can this be done without using RawQuery?

**Reply by grisha-kotler** [reply, marked as answer]
You can use `RavenQuery.Load`.
You can see an example here:
https://github.com/ravendb/ravendb/blob/7beae099fa03721f161489741ad475e9b0aecbf1/test/SlowTests/Issues/RavenDB-13326.cs#L129

**Reply by yjagota** [reply]
Thanks. Exactly what I needed.
